src/coreclr/jit/optimizer.cpp

```diff
@@ -7608,17 +7608,805 @@ bool Compiler::optIsRangeCheckRemovable(GenTree* tree)
     return true;
 }
 
-/******************************************************************************
- *
- * Replace x==null with (x|x)==0 if x is a GC-type.
- * This will stress code-gen and the emitter to make sure they support such trees.
- */
+//
+// OptTestInfo:     Member of OptBoolsDsc struct used to test if a GT-JTRUE or GT_RETURN node
+//                  is a boolean comparison
+//
+struct OptTestInfo
+{
+    GenTree* testTree; // The root node of Basic Block
+    GenTree* compTree; // The compare node (i.e. GT_EQ or GT_NE node) of the testTree
+    bool     isBool;   // If the compTree is boolean comparison
+};
+
+//
+// OptBoolsDsc:     Descriptor used for Boolean Optimization
+//
+struct OptBoolsDsc
```

Comment:
If you've added functions, why not use `class` instead of `struct`?

Reply:
Done.